qiskit/circuit/library/n_local/qaoa_ansatz.py

```diff
@@ -0,0 +1,268 @@
+        valid = True
+
+        if self._cost_operator is None:
+            valid = False
+            if raise_on_failure:
+                raise AttributeError("The operator representing the cost of "
+                                     "the optimization problem is not set")
+        if self._reps is None or self._reps < 1:
+            valid = False
+            if raise_on_failure:
+                raise AttributeError("The integer parameter reps, which determines the depth "
+                                     "of the circuit, needs to be >= 1 but has value {}"
+                                     .format(self._reps))
+
+        if self._initial_state is not None and self._initial_state.num_qubits != self._num_qubits:
+            valid = False
+            if raise_on_failure:
+                raise AttributeError("The number of qubits of the initial state {} does not match "
+                                     "the number of qubits of the cost operator {}"
+                                     .format(self._initial_state.num_qubits, self._num_qubits))
+
+        if self._mixer is not None and self._mixer.num_qubits != self._num_qubits:
+            valid = False
+            if raise_on_failure:
+                raise AttributeError("The number of qubits of the mixer {} does not match "
+                                     "the number of qubits of the cost operator {}"
+                                     .format(self._mixer.num_qubits, self._num_qubits))
+
+        return valid
+
+    def _build(self) -> None:
+        """Build the circuit."""
+        if self._data:
+            return
+
+        self._check_configuration()
+        self._data = []
+
+        # calculate bounds, num_parameters, mixer
+        self._calculate_parameters()
+
+        # parametrize circuit and build it
+        param_vector = ParameterVector("θ", self._num_parameters)
+        circuit = self._construct_circuit(param_vector)
+
+        # append(replace) the circuit to this
+        self.add_register(circuit.num_qubits)
+        self.compose(circuit, inplace=True)
+
+    @property
+    def parameters(self) -> Set[Parameter]:
+        """Get the :class:`~qiskit.circuit.Parameter` objects in the circuit.
+
+        Returns:
+            A set containing the unbound circuit parameters.
+        """
+        self._build()
+        return super().parameters
+
+    def _calculate_parameters(self):
+        """Calculated internal parameters of the circuit to be built."""
+        # self._num_qubits = self._cost_operator.num_qubits
+
+        from qiskit.opflow import OperatorBase
+        if isinstance(self._mixer, QuantumCircuit):
+            self._num_parameters = (1 + self._mixer.num_parameters) * self._reps
+            self._bounds = [(None, None)] * self._reps + \
+                           [(None, None)] * self._reps * self._mixer.num_parameters
+            # self._mixer = self._mixer_operator
```

Comment:
Should we delete these commented lines given its moved - while it was here it seems we could get delete these in the following sections in this method to clean it up

Reply:
Sorry, forgot to delete them.